checks.d/storm_rest_api.py

```diff
@@ -0,0 +1,301 @@
+            task_tags = tags + self.task_id_tags(config, 'spout', name) + [
+                'storm_component_type:spout',
+                'storm_task_id:' + name,
+            ]
+            report_task('spout', name, spout, task_tags)
+            self.gauge(self.metric(config, 'storm.rest.spout.complete_latency_ms'),
+                       float(spout['completeLatency']) * 1000, task_tags)
+
+        for bolt in details.get('bolts'):
+            name = bolt['boltId']
+            task_tags = tags + self.task_id_tags(config, 'bolt', name) + [
+                'storm_component_type:bolt',
+                'storm_task_id:' + name,
+            ]
+            report_task('bolt', name, bolt, task_tags)
+            if bolt['executed'] is not None:
+                executed_count = bolt['executed']
+            else:
+                executed_count = 0
+            self.gauge(self.metric(config, 'storm.rest.bolt.executed_total'),
+                       executed_count, task_tags)
+            self.gauge(self.metric(config, 'storm.rest.bolt.execute_latency_ms'),
+                       float(bolt['executeLatency']) * 1000, task_tags)
+            self.gauge(self.metric(config, 'storm.rest.bolt.process_latency_ms'),
+                       float(bolt['processLatency']) * 1000, task_tags)
+            self.gauge(self.metric(config, 'storm.rest.bolt.capacity_percent'),
+                       float(bolt['capacity']) * 100, task_tags)
+
+    def report_executor_details(self, config, details):
+        """
+        Report statistics for a single topology's task ID's executors.
+        """
+
+        topology_name = self._topology_name(config, details)
+        name = details['id']
+        task_type = details['componentType']
+        tags = config.tags + self.task_id_tags(config, task_type, name) + [
+            'storm_topology:' + topology_name,
+            'storm_component_type:' + task_type,
+            'storm_task_id:' + details['id'],
+        ]
+        self.gauge(self.metric(config, 'storm.rest.executor.executors_total'),
+                               details['executors'], tags=tags)
+        self.gauge(self.metric(config, 'storm.rest.executor.tasks_total'),
+                               details['executors'], tags=tags)
+
+        # TODO: Add tags / process boltStats / spoutStats{?}?
+        if task_type == 'spout':
+            pass
+        elif task_type == 'bolt':
+            pass
+        else:
+            raise "Unknown component type '%s' for '%s'" % (task_type, name)
+
+        # Embarrassingly, executorStats are undocumented in the REST
+        # API docs (so we might not be allowed to rely on them). But
+        # they're the only way to get some SERIOUSLY useful metrics -
+        # per-host metrics, in particular.
+        for executor in details['executorStats']:
+            executor_tags = tags + [
+                'executor_id:' + executor['id'],
+                'storm_host:' + executor['host'],
```

Comment:
Both should have been `storm_host` - it's exactly what you said above (and your comment reminds me that there's a TODO up there cluttering the code).